src/readthedocs-config.js

```diff
@@ -59,63 +61,80 @@ export function getReadTheDocsConfig(sendUrlParam) {
     url = "/_/readthedocs-addons.json";
   }
 
-  return fetch(url, {
-    method: "GET",
-  })
-    .then((response) => {
-      if (!response.ok) {
-        throw "Error parsing configuration data";
-      }
-      return response.json();
+  return url;
+}
+
+/**
+ * Load Read the Docs configuration from API endpoint.
+ *
+ */
+export function getReadTheDocsConfig(sendUrlParam) {
+  return new Promise((resolve, reject) => {
+    let dataUser;
+    const defaultApiUrl = _getApiUrl(sendUrlParam);
+
+    fetch(defaultApiUrl, {
+      method: "GET",
     })
-    .then((data) => {
-      // We force the user to define the `<meta>` tag to be able to use Read the Docs data directly.
-      // This is to keep forward/backward compatibility without breaking integrations.
-      const metadataAddonsAPIVersion = _getMetadataAddonsAPIVersion();
-      if (metadataAddonsAPIVersion !== undefined) {
-        if (metadataAddonsAPIVersion !== data.api_version) {
-          // When the API scheme version returned doesn't match the one defined via `<meta>` tag by the user,
-          // we perform another request to get the Read the Docs response in the structure
-          // that's supported by the user and dispatch a custom event letting them know
-          // this data is ready to be consumed under `event.detail`.
-
-          url =
-            ADDONS_API_ENDPOINT +
-            new URLSearchParams({
-              url: window.location.href,
-              "client-version": CLIENT_VERSION,
-              "api-version": metadataAddonsAPIVersion,
-            });
+      .then((response) => {
+        if (!response.ok) {
+          reject("Error hitting addons API endpoint");
```

Comment:
Noted below, but `reject` doesn't end the function by itself, so this will actually continue on to the `response.json()` below. Usually you add a `return` to force the end of the promise:

```suggestion
          return reject("Error hitting addons API endpoint");
```